### PR TITLE
fix: block no migration

### LINKED ISSUE #12427: (3/27 00:00) Mismatched values, expected XXX, got XXX
### Description

Hashscan [tx](https://hashscan.io/mainnet/transactionsById/0.0.995584-1711497706-539499563):

```
---- EthereumTransaction DIFFERED (TRANSACTION_RECORD_MISMATCH) ----
Mismatched values, expected '0000000000000000000000000000000000000000000000000000000003ae058e000000000000000000000000000000000000000000000000000000000000004000000000000000000000000000000000000000000000000000000000000000010000000000000000000000000000000000000000000000000000000000000020000000000000000000000000000000000000000000000000000000000000000100000000000000000000000000000000000000000000000000000000000dd87e00000000000000000000000000000000000000000000000000000000000000600000000000000000000000000000000000000000000000000000000000000000', got '0000000000000000000000000000000000000000000000000000000003ae058f000000000000000000000000000000000000000000000000000000000000004000000000000000000000000000000000000000000000000000000000000000010000000000000000000000000000000000000000000000000000000000000020000000000000000000000000000000000000000000000000000000000000000100000000000000000000000000000000000000000000000000000000000dd87e00000000000000000000000000000000000000000000000000000000000000600000000000000000000000000000000000000000000000000000000000000000' - Matching field 'contractCallResult' Matching field 'contractCallResult'  ==> expected: <<ByteString@4f7c0be3 size=256 contents="\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\003\256\005\216\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000...">> but was: <<ByteString@7e0b9178 size=256 contents="\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\003\256\005\217\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000...">>

For body,
transactionID {
  transactionValidStart {
    seconds: 1711497706
    nanos: 539499563
  }
  accountID {
    accountNum: 995584
  }
}
nodeAccountID {
  accountNum: 4
}
transactionFee: 1080000000
transactionValidDuration {
  seconds: 120
}
ethereumTransaction {
  ethereum_data: "\002\371\001\267\202\001\'\202\b\302\205\326\347\026N\000\205\326\347\026N\000\203\024\223 \224\362#\231a\322\351\026P?\323S\025f\a3\270\243#\271)\200\271\001D\027I\341\343\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000 \000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\001\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000 \000\000\000\000\000\000\000\000\000\000\000\000>\212Y\247\353N\253\324\030\324D\024;\034\221\327m\2448N\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\023\317\320\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000`\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000DE\205\343;\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000 \000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\000\300\200\240F\202p\365\022~\235\351v\344\205\'$\002\022\223\033\023\275O\364`f\216\332\204\356R\202X\266\302\240\'\370\310\361tC\r\036\004\020;\361\b\177\t`\251\203\300\026\345K)\214\301\206\"\327\366\223\277\377"
}
➡️  Expected Record receipt {
  status: SUCCESS
```

### Steps to reproduce

Run 15 mins events on round 166444130, 166445137, 166447149

### Additional context

_No response_

### Hedera network

other

### Version

release/0.48

### Operating system

None

### PR DESCRIPTION
**Pull request by tinker-michaelj**
**Description**:
 - Closes #12427 
 - Migrates last block number to `(current - 1)` from mono state.

### CODE REVIEW

**Review by iwsimon — DISMISSED** [review]
LGTM, thanks @tinker-michaelj !

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/records/impl/codec/BlockInfoTranslator.java [hunk 1]
```diff
@@ -43,7 +43,7 @@ public static BlockInfo blockInfoFromMerkleNetworkContext(
             throws IOException {
         requireNonNull(merkleNetworkContext);
         final var blockInfoBuilder = new BlockInfo.Builder()
-                .lastBlockNumber(merkleNetworkContext.getAlignmentBlockNo())
+                .lastBlockNumber(merkleNetworkContext.getAlignmentBlockNo() - 1)
```
Comment (david-bakin-sl) [inline hedera-node/hedera-app/src/main/java/com/hedera/node/app/records/impl/codec/BlockInfoTranslator.java:46]: This arbitrary-seeming `-1` will possibly puzzle the future reader.  Perhaps a comment to indicate what it's for?  (I mean, I know its to fix a diff testing issue, but why `-1` and not `+0`?)

**Review by david-bakin-sl — COMMENTED** [review]
Magic subtraction will be a mystery to future readers.  Suggest adding a comment.

**Review by povolev15 — APPROVED** [review]

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/records/impl/codec/BlockInfoTranslator.java (hunk 1 above)
Comment (tinker-michaelj) [reply hedera-node/hedera-app/src/main/java/com/hedera/node/app/records/impl/codec/BlockInfoTranslator.java:49]: Added comment!

**Review by Neeharika-Sompalli — APPROVED** [review]

**Pull request merged**